test/distributed/org/apache/cassandra/distributed/test/SchemaTest.java

```diff
@@ -197,4 +203,114 @@ private static boolean checkTablesPropagated(IInvokableInstance instance, boolea
                    && (Schema.instance.getTableMetadata(KEYSPACE, TABLE_TWO) != null ^ !two);
         });
     }
+
+    @Test
+    public void dropColumnMixedMode() throws Throwable
+    {
+        try (Cluster cluster = init(Cluster.build(2).start()))
+        {
+            cluster.schemaChange(withKeyspace("CREATE TABLE %s.tbl (id int primary key, v1 int, v2 int, v3 int)"));
+            cluster.coordinator(1).execute(withKeyspace("INSERT INTO %s.tbl (id, v1, v2, v3) VALUES (?,?,?, ?)") , ConsistencyLevel.ALL, 1, 10, 100, 1000);
+            cluster.coordinator(1).execute(withKeyspace("INSERT INTO %s.tbl (id, v1, v2, v3) VALUES (?,?,?, ?)") , ConsistencyLevel.ALL, 2, 20, 200, 2000);
+            cluster.coordinator(1).execute(withKeyspace("INSERT INTO %s.tbl (id, v1, v2, v3) VALUES (?,?,?, ?)") , ConsistencyLevel.ALL, 3, 30, 300, 3000);
+            cluster.coordinator(1).execute(withKeyspace("INSERT INTO %s.tbl (id, v1, v2, v3) VALUES (?,?,?, ?)") , ConsistencyLevel.ALL, 4, 40, 400, 4000);
+
+            cluster.forEach((instance) -> instance.flush(KEYSPACE));
+            cluster.get(1).schemaChangeInternal(withKeyspace("ALTER TABLE %s.tbl DROP v2"));
+
+            assertRows(cluster.coordinator(1).execute(withKeyspace("SELECT id, v1 FROM %s.tbl"), ConsistencyLevel.ALL),
+                       row(1, 10),
+                       row(2, 20),
+                       row(4, 40),
+                       row(3, 30));
+
+            assertRows(cluster.coordinator(1).execute(withKeyspace("SELECT * FROM %s.tbl"), ConsistencyLevel.ALL),
+                       row(1, 10, 1000),
+                       row(2, 20, 2000),
+                       row(4, 40, 4000),
+                       row(3, 30, 3000));
+
+            assertRows(cluster.coordinator(2).execute(withKeyspace("SELECT id, v1 FROM %s.tbl"), ConsistencyLevel.ALL),
+                       row(1, 10),
+                       row(2, 20),
+                       row(4, 40),
+                       row(3, 30));
+
+            assertRows(cluster.coordinator(2).execute(withKeyspace("SELECT * FROM %s.tbl"), ConsistencyLevel.ALL),
+                       row(1, 10, 100, 1000),
+                       row(2, 20, 200, 2000),
+                       row(4, 40, 400, 4000),
+                       row(3, 30, 300, 3000));
```

Comment:
How about propagating the schema change and checking nodes 1 and 2 see the same then?

Reply:
There is apparently no way to let the schema propagate. We will have to apply the same schema change to the other node which is the same that what we did for the first node.